kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -549,6 +547,8 @@ private SnapshotImpl createSnapshot(LogSegment initSegment, Engine engine) {
         initSegment.version,
         startingFromStr);
 
+    assertLogFilesBelongToTable(logPath, initSegment.allLogFilesUnsorted());
```

Comment:
I believe the main goal of the log statement above is to print out how long it took to load the protocol and metadata objects during LogReplay.

So I think the easiest thing to do would be to update the log statement (to not directly mention constructing the snapshot) and move it to right after the LogReplay instance is created. what do you think?